tools/devops/automation/templates/packages/build.yml

```
@@ -0,0 +1,454 @@
+  clean: true             # Executes: git clean -ffdx && git reset --hard HEAD
+  submodules: recursive
+
+- checkout: maccore
+  clean: true
+  persistCredentials: true  # hugely important, else there are some scripts that check a single file from maccore that will fail
+
+- checkout: templates
+  clean: true
+
+- checkout: release-scripts
+  clean: true
+
+- powershell: |
+    gci env: | format-table -autosize -wrap
+  displayName: 'Dump Environment'
+
+- bash: $(System.DefaultWorkingDirectory)/xamarin-macios/tools/devops/automation/scripts/bash/clean-bot.sh
+  displayName: 'Clean bot'
+  env:
+    BUILD_REVISION: 'jenkins'
+
+- bash: |
+    security set-key-partition-list -S apple-tool:,apple: -s -k $OSX_KEYCHAIN_PASS login.keychain
+  env:
+    OSX_KEYCHAIN_PASS: $(OSX_KEYCHAIN_PASS)
+  displayName: 'Remove security UI-prompt (http://stackoverflow.com/a/40039594/183422)'
+  condition: succeededOrFailed() # we do not care about the previous process cleanup
+
+- template: install-certificates.yml@templates
+  parameters:
+    DeveloperIdApplication: $(developer-id-application)
+    DeveloperIdInstaller: $(developer-id-installer)
+    IphoneDeveloper: $(iphone-developer)
+    MacDeveloper: $(mac-developer)
+    HostedMacKeychainPassword: $(OSX_KEYCHAIN_PASS)
+
+- task: xamops.azdevex.provisionator-task.provisionator@2
+  displayName: 'Provision Brew components'
+  inputs:
+    provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-brew-packages.csx
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 30
+  enabled: false
+
+- bash: |
+    make -C $(Build.SourcesDirectory)/xamarin-macios/tools/devops build-provisioning.csx
+  displayName: 'Generate provisionator files.'
+
+- task: xamops.azdevex.provisionator-task.provisionator@1
+  displayName: 'Provision Products & Frameworks'
+  inputs:
+    provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/build-provisioning.csx
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 250
+
+- task: xamops.azdevex.provisionator-task.provisionator@1
+  displayName: 'Provision .NET Core'
+  inputs:
+    provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-dotnetcore.csx
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 30
+
+- powershell: |
+    if ([IO.Directory]::Exists('/Developer/MonoTouch')) {
+      Write-Host "Deleting /Developer/MonoTouch"
+      bash -c "sudo rm -Rf /Developer/MonoTouch"
+    }
+
+    if ([IO.Directory]::Exists('/Library/Frameworks/Xamarin.iOS.framework')) {
+      Write-Host "Deleting /Library/Frameworks/Xamarin.iOS.framework"
+      bash -c "sudo rm -Rf /Library/Frameworks/Xamarin.iOS.framework"
+    }
+
+    if ([IO.Directory]::Exists('/Library/Frameworks/Xamarin.Mac.framework')) {
+      Write-Host "Deleting /Library/Frameworks/Xamarin.Mac.framework"
+      bash -c "sudo rm -Rf /Library/Frameworks/Xamarin.Mac.framework"
+    }
```

This seems a bit simpler:

```suggestion
- bash: |
    set -x
    sudo rm -Rf /Developer/MonoTouch
    sudo rm -Rf /Library/Frameworks/Xamarin.iOS.framework
    sudo rm -Rf /Library/Frameworks/Xamarin.Mac.framework
```